bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartImpl.java

```diff
@@ -318,9 +335,12 @@ public void dispose() {
         }
         httpClient.destroy();
         webSocketClient.destroy();
-        locationsResponse = new LocationsResponse();
+        LocationsResponse locationsResponse = this.locationsResponse;
+        if (locationsResponse != null && locationsResponse.data != null) {
+            locationsResponse.data.clear();
+        }
```

Comment:
What's the point of clearing that data member if the whole object is discarded below anyway?

Reply:
I changed the code, but your comments on inner workings of Java garbage collector would still be appreciated.